mlx/backend/metal/indexing.cpp

```diff
@@ -641,4 +643,154 @@ void ScatterAxis::eval_gpu(const std::vector<array>& inputs, array& out) {
   compute_encoder.dispatch_threads(grid_dims, group_dims);
 }
 
+void MaskedScatter::eval_gpu(const std::vector<array>& inputs, array& out) {
+  const array& self = inputs[0];
```

Comment:
This is a nit but I think it makes sense to change the name `self` to `out` (or `dst` or something similar). The term `self` is a bit overloaded from Python so it's slightly confusing.